lib/asset/assetversion.js

```diff
@@ -28,10 +29,10 @@ class AssetVersion {
      */
     constructor(lastModified, etag) {
         if (lastModified && !isValidNumber(lastModified)) {
-            throw Error(`Invalid lastModified: ${lastModified}`);
+            throw new IllegalArgumentError("Invalid lastModified", lastModified);
```

Comment:
I would expect `lastModified` to possibly also be a date? Or is everything converted to timestamps?